Support for the regex_search_ternary used by the realtime profile.

See: https://bugzilla.redhat.com/show_bug.cgi?id=1797025

--- /dev/null
+++ b/usr/lib/python2.7/site-packages/tuned/profiles/functions/function_regex_search_ternary.py
@@ -0,0 +1,21 @@
+import re
+from . import base
+
+class regex_search_ternary(base.Function):
+	"""
+	Ternary regex operator, it takes arguments in the following form
+	STR1, REGEX, STR2, STR3
+	If REGEX matches STR1 (re.search is used), STR2 is returned,
+	otherwise STR3 is returned
+	"""
+	def __init__(self):
+		# 4 arguments
+		super(regex_search_ternary, self).__init__("regex_search_ternary", 4, 4)
+
+	def execute(self, args):
+		if not super(regex_search_ternary, self).execute(args):
+			return None
+		if re.search(args[1], args[0]):
+			return args[2]
+		else:
+			return args[3]
